server/core/src/https/views/enroll.rs

```diff
@@ -0,0 +1,115 @@
+use askama::Template;
+use askama_axum::IntoResponse;
+
+use axum::extract::State;
+use axum::response::Response;
+use axum::Extension;
+
+use axum_extra::extract::CookieJar;
+use kanidm_proto::internal::UserAuthToken;
+
+use qrcode::render::svg;
+use qrcode::QrCode;
+use url::Url;
+
+use std::time::Duration;
+
+use super::constants::Urls;
+use super::navbar::NavbarCtx;
+use crate::https::extractors::{DomainInfo, VerifiedClientInformation};
+use crate::https::middleware::KOpId;
+use crate::https::views::constants::ProfileMenuItems;
+use crate::https::views::errors::HtmxError;
+use crate::https::views::login::{LoginDisplayCtx, Reauth, ReauthPurpose};
+use crate::https::ServerState;
+
+#[derive(Template)]
+#[template(path = "user_settings.html")]
+struct ProfileView {
+    navbar_ctx: NavbarCtx,
+    profile_partial: EnrollDeviceView,
+}
+
+#[derive(Template)]
+#[template(path = "enroll_device.html")]
+pub(crate) struct EnrollDeviceView {
+    menu_active_item: ProfileMenuItems,
+    secret: String,
+    qr_code_svg: String,
+    uri: Url,
+}
+
+pub(crate) async fn view_enroll_get(
+    State(state): State<ServerState>,
+    Extension(kopid): Extension<KOpId>,
+    VerifiedClientInformation(client_auth_info): VerifiedClientInformation,
+    DomainInfo(domain_info): DomainInfo,
+    jar: CookieJar,
+) -> axum::response::Result<Response> {
+    let uat: UserAuthToken = state
+        .qe_r_ref
+        .handle_whoami_uat(client_auth_info.clone(), kopid.eventid)
+        .await
+        .map_err(|op_err| HtmxError::new(&kopid, op_err))?;
+
+    let time = time::OffsetDateTime::now_utc() + time::Duration::new(60, 0);
+    let can_rw = uat.purpose_readwrite_active(time);
+
+    if can_rw {
```

Comment:
Would be nice to `!can_rw` and remove the else. Save some whitespace with the bigger block